libkbfs/key_manager_test.go

```diff
@@ -1077,6 +1077,92 @@ func TestKeyManagerReaderRekey(t *testing.T) {
 	}
 }
 
+func TestKeyManagerReaderRekeyAndRevoke(t *testing.T) {
```

Comment:
Actually the first rekey should be from u2d1, and the second can be from either u2d1 or u2d2 and the result should be the same.

I agree, the case you linked should probably return `RekeyIncompleteError`.